src/az_iot_hub_client.c

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+
+#include <az_result.h>
+#include <az_span.h>
+#include <az_version.h>
+#include <az_precondition_internal.h>
+#include <az_result_internal.h>
+#include <az_span_internal.h>
+#include <az_iot_hub_client.h>
+#include <az_iot_common_internal.h>
+
+#include <_az_cfg.h>
+
+static const uint8_t null_terminator = '\0';
+static const uint8_t hub_client_forward_slash = '/';
+static const az_span hub_client_param_separator_span = AZ_SPAN_LITERAL_FROM_STR("&");
+static const az_span hub_client_param_equals_span = AZ_SPAN_LITERAL_FROM_STR("=");
+
+static const az_span hub_digital_twin_model_id = AZ_SPAN_LITERAL_FROM_STR("model-id");
+static const az_span hub_service_api_version = AZ_SPAN_LITERAL_FROM_STR("/?api-version=2020-09-30");
+static const az_span client_sdk_device_client_type_name = AZ_SPAN_LITERAL_FROM_STR("dct");
+static const az_span client_sdk_version_default_value
+    = AZ_SPAN_LITERAL_FROM_STR("azsdk-c%2F" AZ_SDK_VERSION_STRING);
```

Comment:
Yes, that's exactly what src/ has. The latest code from main@azure-sdk-for-c. That is what we planned originally. Should it be different?

Reply:
Picking an arbitrary point-in-time commit sha from the main repo is probably not the best idea. Though unlikely, it might have some stability or other product bug. Can we pick an existing released tag instead?

Or is the intention to stay in alignment with main, and then before we officially ship a version, we produce a new release/tag from main, and update this to the sources at that point?

Reply:
Discussed offline and we will use 1.3.0-beta.1 for this first release.

Reply:
As we talked we will use azure-sdk-for-c version 1.3.0-beta.1.